### PR TITLE
feat(radio)!: remove top level subcomponents

### PR DESCRIPTION
**Pull request by jinlee93**
[EFI-1231]

### Summary:
- removes radio subcomponents

### Test Plan:

- [x] CI tests / new tests are not applicable
  - no visual regressions

[EFI-1231]: https://czi-tech.atlassian.net/browse/EFI-1231?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/Radio/Radio.module.css [hunk 1]
```diff
@@ -9,3 +9,88 @@
...
+}
+/**
+ * The visually hidden input element for the radio. The visual radio is provided by an svg element.
+ */
+.radio__input {
+  /* Removes default margins placed by browser for radioes. */
+  margin: 0;
+  /* Remove the radio from the page flow, positioning it on top of the SVG. */
+  position: absolute;
+  left: 0.25rem;
```
Comment (booc0mtaco) [inline src/components/Radio/Radio.module.css:32]: nit: this should be `--eds-size-half` i think but I'm guessing this was just brought over from the other file

File: src/components/Radio/Radio.module.css [hunk 2]
```diff
@@ -9,3 +9,88 @@
...
+/**
+ * Radio label size variants.
+ * Used for centering the label with the radio input button.
+ */
+.radio__label--md {
+  top: 0.125rem;
+}
+.radio__label--lg {
+  top: 0.0625rem;
+}
```
Comment (booc0mtaco) [inline src/components/Radio/Radio.module.css:96]: Not a fan of this, but again this was just brought over :)

Comment (booc0mtaco) [reply src/components/Radio/Radio.module.css:96]: 0.125rem corresponds to `--eds-size-quarter` which does exist, but `--eds-size-eighth` or what have you does not...and probably shouldn't. the sizes tokens are kinda weird. I wonder if there are other cases of such a small relative spacing that would indicate we should add this token?

Comment (jinlee93) [reply src/components/Radio/Radio.module.css:96]: image
There are a some instances where `0.0625rem` is used; still tho `eighth` is kind of small?

Comment (booc0mtaco) [reply src/components/Radio/Radio.module.css:96]: very small... one pixel in fact :/ 

This seems to exist to vertically center the text with the radio button. there may be a way to do the same via flex and avoid this kinda thing altogether. So as not to over-complicate this PR, we can make a comment on why such a precise value is being used, then leave it be for now

Comment (jinlee93) [reply src/components/Radio/Radio.module.css:96]: That makes sense to me, will do

**Pull request merged**